chromadb/test/property/test_embeddings.py

```diff
@@ -138,18 +138,24 @@ def add_embeddings(self, record_set: strategies.RecordSet) -> MultipleResults[ID
             }
             self.collection.add(**normalized_record_set)  # type: ignore[arg-type]
             self._upsert_embeddings(cast(strategies.RecordSet, filtered_record_set))
-            return multiple(*filtered_record_set["ids"])
+            return multiple(*filtered_record_set["ids"] or [])
 
         else:
-            self.collection.add(**normalized_record_set)  # type: ignore[arg-type]
+            result = self.collection.add(**normalized_record_set)  # type: ignore[arg-type]
+
+            if normalized_record_set["ids"] is None:
+                normalized_record_set["ids"] = result["ids"]
+
             self._upsert_embeddings(cast(strategies.RecordSet, normalized_record_set))
-            return multiple(*normalized_record_set["ids"])
+            return multiple(*normalized_record_set["ids"] or [])
 
     @rule(ids=st.lists(consumes(embedding_ids), min_size=1))
     def delete_by_ids(self, ids: IDs) -> None:
         trace("remove embeddings")
         self.on_state_change(EmbeddingStateMachineStates.delete_by_ids)
-        indices_to_remove = [self.record_set_state["ids"].index(id) for id in ids]
+
+        normalized_ids = self.record_set_state["ids"] or []
```

Comment:
By the time we arrive at `delete` we should know the ids of what we intend to delete